apps/api/src/tasks/bank-import/import-transactions.task.ts

```diff
@@ -298,8 +307,23 @@ export class IrisTasks {
         .replace(/[ _]+/g, '-')
         .match(this.regexPaymentRef)
 
+      const transactionAmount = {
+        amount: trx.transactionAmount?.amount,
+        currency: trx.transactionAmount?.currency,
+      }
+      const id = trx.transactionId?.trim() || ''
+
+      // If we receive EUR in the BGN account - try parsing from the transaction id the amount in BGN
+      if (trx.transactionAmount?.currency === Currency.EUR && trx.transactionAmount?.amount > 0) {
```

Comment:
@igoychev  Sure, I can do that. I'm worried that we don't know if the same parsing rule will apply + I guess we haven't seen any != EUR/BGN transactions, so we might want to actually see these and then handle them correctly? I'm worried that this might hide some issues and automatically "resolve" them. Let me know what you think

Reply:
Well, because the receiving account is BGN then it is safe to assume, that the amount coming in the transactionId is always BGN for all types of incoming currencies.